quick_start/sample_app.py

```diff
@@ -0,0 +1,350 @@
+import enum
+import json
+import uuid
+from dataclasses import dataclass
+import inflection as inflection
+from azure.kusto.ingest import QueuedIngestClient
+from quick_start.utils import Utils, AuthenticationModeOptions
+from azure.kusto.data import DataFormat, KustoClient
+
+
+class SourceType(enum.Enum):
+    """
+    SourceType - represents the type of files used for ingestion
+    """
+
+    local_file_source = "localFileSource"
+    blob_source = "blobSource"
+
+
+class ConfigData:
+    """
+    ConfigData object - represents a file from which to ingest
+    """
+
+    source_type: SourceType
+    data_source_uri: str
+    data_format: DataFormat
+    use_existing_mapping: bool
+    mapping_name: str
+    mapping_value: str
+
+
+@dataclass
+class ConfigJson:
+    """
+    ConfigJson object - represents a cluster and DataBase connection configuration file.
+    """
+
+    use_existing_table: bool
+    database_name: str
+    table_name: str
+    table_schema: str
+    kusto_uri: str
+    ingest_uri: str
+    data_to_ingest: list[ConfigData]
+    alter_table: bool
+    query_data: bool
+    ingest_data: bool
+    authentication_mode: AuthenticationModeOptions
+    wait_for_user: bool
+    wait_for_ingest_seconds: bool
+    batching_policy: str
+
+
+class KustoSampleApp:
+    # TODO (config):
+    #  If this quickstart app was downloaded from OneClick, kusto_sample_config.json should be pre-populated with your cluster's details
+    #  If this quickstart app was downloaded from GitHub, edit kusto_sample_config.json and modify the cluster URL and database fields appropriately
+    CONFIG_FILE_NAME = "kusto_sample_config.json"
+
+    __step = 1
+    config = None
+
+    @classmethod
+    def load_configs(cls, config_file_name: str) -> None:
+        """
+        Loads JSON configuration file, and sets the metadata in place.
+        :param config_file_name: Configuration file path.
+        """
+        try:
+            with open(config_file_name, "r") as config_file:
+                json_dict = json.load(config_file)
+
+        except Exception as ex:
+            Utils.error_handler(f"Couldn't read load config file from file '{config_file_name}'", ex)
+
+        json_dict["dataToIngest"] = cls.convert_config_data_fields(json_dict["data"])
+        json_dict.pop("data", None)
+        cls.config = ConfigJson(**{inflection.underscore(key): val for (key, val) in json_dict.items()})
+
+    @classmethod
+    def convert_config_data_fields(cls, data_dicts: list[dict]) -> list[ConfigData]:
+        """
+        Converts dict object  - JSON style camelCase to ConfigData object - python snake_case
+        :param data_dicts: Dict of data sources list to ingest from - styled in camelCase
+        :return: ConfigData Data sources list to ingest from - styled in snake_case
+        """
+        config_data_list = []
+
+        for data_dict in data_dicts:
+            config_data = ConfigData()
```

Comment:
I know what we discussed in the chat, but don't you think this is nicer?
```python
            data_dict["sourceType"] = SourceType[inflection.underscore(data_dict.pop("sourceType"))]
            data_dict["dataFormat"] = DataFormat[data_dict.pop("format")]
            config_data_list.append(ConfigData(**KustoSampleApp.keys_to_snake_case(data_dict)))
```

If not, you can keep this as is.